az-core/src/main/java/azkaban/metrics/MetricsManager.java

```diff
@@ -117,14 +117,19 @@ public Timer addTimer(final String name) {
   public synchronized void startReporting(final String reporterName, final Props props) {
     final String metricsReporterClassName = props.get(CUSTOM_METRICS_REPORTER_CLASS_NAME);
     final String metricsServerURL = props.get(METRICS_SERVER_URL);
+    final String customHostNameToEmitMetrics = props.get(CUSTOM_METRICS_HOSTNAME);
```

Comment:
What's the behavior when the property `CUSTOM_METRICS_HOSTNAME` is not set ?
Ideally we would wan't the default behavior to stay unchanged, if that's the case please mention that in a comment. Please let me know if you think otherwise.

Reply:
this property is an optional one. If we don't set it, the default behaviour stays as it is. Let me update the PR description too. Thanks for taking a look!